role-service/src/main/java/uk/ac/ox/ndph/mts/role_service/service/RoleService.java

```diff
@@ -1,26 +1,55 @@
 package uk.ac.ox.ndph.mts.role_service.service;
 
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 import uk.ac.ox.ndph.mts.role_service.controller.DuplicateRoleException;
+import uk.ac.ox.ndph.mts.role_service.model.Permission;
+import uk.ac.ox.ndph.mts.role_service.model.PermissionRepository;
 import uk.ac.ox.ndph.mts.role_service.model.Role;
 import uk.ac.ox.ndph.mts.role_service.model.RoleRepository;
 
+import java.util.List;
+import java.util.Optional;
+
 @Service
 public class RoleService {
 
 
     private final RoleRepository roleRepository;
+    private final PermissionRepository permissionRepository;
 
-    public RoleService(RoleRepository roleRepository) {
+    public RoleService(RoleRepository roleRepository, PermissionRepository permissionRepository) {
         this.roleRepository = roleRepository;
+        this.permissionRepository = permissionRepository;
     }
 
     public Role saveRole(Role role) throws DuplicateRoleException {
 
         if (roleRepository.existsById(role.getId())) {
             throw new DuplicateRoleException();
         }
+        return roleRepository.save(role);
+    }
+
+    public Role updatePermissionsForRole(String roleId, List<Permission> newPermissions)
+            throws ResponseStatusException {
+
+        Optional<Role> roleOptional = roleRepository.findById(roleId);
+        if (roleOptional.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Role \"" + roleId + "\" not found.");
```

Comment:
In other services we used enums + string format for those. 
An example: https://github.com/NDPH-ARTS/mts-services/blob/main/practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/FhirRepo.java

Reply:
Thanks @tamirkamara good point.  I've refactored to an enum.